src/poetry/console/application.py

```diff
@@ -342,6 +344,15 @@ def _load_plugins(self, io: IO | None = None) -> None:
 
         self._plugins_loaded = True
 
+    @staticmethod
+    def _load_system_truststore() -> None:
+        from poetry.utils.ssl_truststore import is_truststore_enabled
+
+        if is_truststore_enabled():
+            import truststore
+
+            truststore.inject_into_ssl()
```

Comment:
shouldn't this be wrapped in try/catch for ImportError? We should not need the `ssl_truststore` module.

Reply:
As you can see, there are several checks done in `ssl_truststore` module. I wanted to isolate it and avoid cluttering the `Application` with it.

Reply:
As I understood it those checks are redundant. If truststore cannot be used an import error will be raised.

Reply:
Simple `ImportError` is too little in my taste. The `_is_truststore_available()` function has various checks and logs a proper message.

Reply:
It might be, but it does not add anything in this context.

https://truststore.readthedocs.io/en/latest/

> If Truststore can’t work for a given platform due to APIs not being available then at import time the exception ImportError will be raised with an informative message

The only case where it makes sense is if we need to programmatically check if truststore is available. If that is the case, we might not want `inject_into_ssl` anyway.

Also note;
> The call to truststore.inject_into_ssl() should be called as early as possible in your program as modules that have already imported ssl.SSLContext won’t be affected.

This means this might be too late as imports to repos and authenticator would have already happened.